app/lib/dns.server.ts

```diff
@@ -0,0 +1,200 @@
+import {
+  Route53Client,
+  CreateHostedZoneCommand,
+  ChangeResourceRecordSetsCommand,
+  GetChangeCommand,
+} from "@aws-sdk/client-route-53";
+import {
+  CreateHostedZoneResponse,
+  ChangeResourceRecordSetsResponse,
+  GetChangeResponse,
+} from "@aws-sdk/client-route-53";
+import { isIPv4, isIPv6 } from "is-ip";
+
+export const route53Client = new Route53Client({
+  endpoint: process.env.AWS_ENDPOINT_URL || "http://localhost:5053",
+  region: process.env.AWS_REGION || "us-east-1",
+  credentials: {
+    accessKeyId: process.env.AWS_ACCESS_KEY_ID || "",
+    secretAccessKey: process.env.AWS_SECRET_ACCESS_KEY || "",
+    sessionToken: process.env.AWS_SESSION_TOKEN,
+  },
+});
+
+export const createHostedZone = async (domain: string) => {
+  try {
+    const command = new CreateHostedZoneCommand({
+      Name: domain,
+      CallerReference: new Date().toString(),
+    });
+    const response: CreateHostedZoneResponse = await route53Client.send(
+      command
+    );
+
+    if (!response.HostedZone?.Id) {
+      throw new Error("Missing hosted zone ID in AWS response");
+    }
+    return response.HostedZone?.Id.replace("/hostedzone/", "");
```

Comment:
Drop the `?` here, it's not needed, due to check in 34.

I already mentioned all this in previous reviews.